### PR TITLE
Viewer controls

### PR DESCRIPTION
**Pull request by gusferguson**
@jburel 

I have also added in a section on the color picker dialog.

Work in progress

### CODE REVIEW

**Comment by gusferguson** [comment]
@jburel 

This is what the table layout looked like:

### General controls

| Control | State | Condition |
| ------- | ----- | --------- |
| Save | Disabled | default |
| | | user permissions exclude action |
| | Enabled | user permissions allow |
| | | when the settings have been changed |
| | At Event | check against settings prior to action |
| | Action | saves settings for current image and current user |
| Save to All | Disabled | user permissions exclude action |
| | Enabled | default - if user permissions allow |
| | At Event | check against settings prior to action |
| | Action | saves settings for current image and current user to all images in dataset |

**Comment by jburel** [comment]
Maybe more "intimidating", if we want this type of document to become the norm.
Leave it as it is for now, we can alway revisit

File: viewer_controls.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+**Copy**
+- Enabled
+ - by default
+- Action
+ - copy the rendering settings
+
+**Paste**
+- Disabled
+ - default
+ - for the image the settings are copied from
```
Comment (will-moore) [inline viewer_controls.md]: We allow Paste onto the same image that we Copied from. E.g. copy settings you like, make some changes, paste to undo changes.

File: viewer_controls.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+- Enabled
+ - when saved settings available to paste
```
Comment (will-moore) [inline viewer_controls.md]: Also, copied settings must be compatible with current image - Same number of channels and same pixel type.

File: viewer_controls.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+- At Event
+ - check against settings prior to action
+- Action
+ - undo the last rendering change
```
Comment (will-moore) [inline viewer_controls.md]: Undo? when Pasting? Should this line be removed?

File: viewer_controls.md (hunk 2 above)
Comment (jburel) [inline viewer_controls.md]: That is taken care by the server.

File: viewer_controls.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+- At Event
+ - open color-picker dialog
+
+#### Color picker dialog
+
+**Set-up: LUT loading**
+- lut are loaded using the method ``listLuts_json`` see [views.py](<https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/views.py>)
+- currently the icon associated to a lut is determined using a pre-build image. (To be modified)
+
+**Toolbar button - color wheel**
```
Comment (will-moore) [inline viewer_controls.md]: The color-picker UI is a bit different in web. No toolbar buttons etc.

File: viewer_controls.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+#### Controls for all channels
+
+**Greyscale checkbox**
+- Unchecked
+ - default
+- Event - check
+ - toggles all on channels off
+- set rendering to grey scale
+- Event - uncheck
+ - toggles all channels that were on back on
```
Comment (will-moore) [inline viewer_controls.md]: In web, we don't keep track of what channels were on before greyscale, so we still only have a single channel on after reverting greyscale -> color.

File: viewer_controls.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+### Users settings
+
+**User setting thumbnail and text**
+- thumbnail displayed is generated using the last settings saved by the users who viewed the image
+- current implementation
+ - one thumbnail per user
+- text displayed under the thumbnail
+ - from string composed of the first name and last name of the user who viewed the image and saved settings
+- thumbnail selected
+ - all settings re-initialized
```
Comment (will-moore) [inline viewer_controls.md]: we add a blue border around the clicked thumbnail that remains there until user saves settings - then blue border reverts to the user's own thumbnail.

**Pull request merged**